client/www/lib/hooks/useLocalStorage.tsx

```diff
@@ -0,0 +1,48 @@
+import { useCallback, useRef, useSyncExternalStore } from 'react';
+
+function getSnapshot<T>(k: string): T | undefined {
+  if (typeof window == 'undefined') return;
+  let v = window.localStorage.getItem(k);
+  if (!v) return;
+  try {
+    return JSON.parse(v);
+  } catch (e) {}
+}
+
+function getServerSnapshot(k: string): undefined {
+  return;
+}
+
+function setItem<T>(k: string, v: T | undefined) {
+  if (typeof window == 'undefined') {
+    throw new Error('useLocalStorage/setState needs to run on the client');
+  }
+  const stringified = JSON.stringify(v);
+  window.localStorage.setItem(k, stringified);
+  // localStorage.setItem does not dispatch events to the current
+  window.dispatchEvent(
+    new StorageEvent('storage', { key: k, newValue: stringified })
+  );
+}
+
+export default function useLocalStorage<T = {}>(
```

Comment:
made a nifty new 'useLocalStorage` hook! Could be with stuff like 'active tab'. Maybe one day can be replaced with `db.local`!